src/UnderlineNav2/UnderlineNav.tsx

```diff
@@ -321,26 +296,30 @@ export const UnderlineNav = forwardRef(
           theme,
           setChildrenWidth,
           setNoIconChildrenWidth,
-          selectedLink,
-          setSelectedLink,
-          selectedLinkText,
-          setSelectedLinkText,
-          selectEvent,
-          afterSelect: afterSelectHandler,
-          variant,
           loadingCounters,
           iconsVisible,
         }}
       >
         {ariaLabel && <VisuallyHidden as="h2">{`${ariaLabel} navigation`}</VisuallyHidden>}
         <Box
           as={as}
-          sx={merge<BetterSystemStyleObject>(getNavStyles(theme, {align}), sxProp)}
+          sx={merge<BetterSystemStyleObject>(getNavStyles(theme), sxProp)}
           aria-label={ariaLabel}
           ref={navRef}
         >
           <NavigationList sx={ulStyles} ref={listRef} role="list">
-            {responsiveProps.items}
+            {listLinks.map(listLink => {
+              return (
+                <Box
```

Comment:
I moved the Box wrapper here to make UnderlineNav return `<Link></Link>` instead of `<Box><Link></Link></Box` to align with `as PolymorphicForwardRefComponent<'a', UnderlineNavItemProps>` but let me know if you have any concern. I am not 100% sure about that.